src/herder/TransactionQueue.cpp

```diff
@@ -752,11 +579,12 @@ TransactionQueue::ban(Transactions const& banTxs)
 
     // Group the transactions by source account and ban all the transactions
     // that are explicitly listed
-    std::map<AccountID, Transactions> transactionsByAccount;
+    std::map<AccountID, TransactionFrameBasePtr> transactionsByAccount;
     for (auto const& tx : banTxs)
     {
-        auto& transactions = transactionsByAccount[tx->getSourceID()];
-        transactions.emplace_back(tx);
+        // Must be a new transaction for this account
+        releaseAssert(
```

Comment:
This condition is new and not trivial. Are we sure we never trigger this assertion?

Reply:
Actually, it doesn't make things that much better, since we need to construct maps and validate that keys map to the right transactions. I left this assert as-is - I audited the code and confirmed that we only ban txs with unique source accounts.